[Build] Remove configure-xcode-for-embedded-development check in symlinking script

emw-apple opened the pull request:
#### 91f009ce1fe5401a0430f0274a88d6a5f758803b
<pre>
[Build] Remove configure-xcode-for-embedded-development check in symlinking script
<a href="https://rdar.apple.com/132780726">rdar://132780726</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=277326">https://bugs.webkit.org/show_bug.cgi?id=277326</a>

Reviewed by Alexey Proskuryakov.

Previously, this script checked for an Xcode installation that had been
modified by configure-xcode-for-embedded-development, and would re-use
those modified headers if needed. Since it has been a year since this
behavior change, and anyone buliding WebKit for embedded platforms needs
to be on a recent SDK version, this is no longer needed.

Instead, repurpose the check to detect when a header of framework we&apos;re
copying on top of the public SDK is no longer needed. Have WTF emit an
error indicating which headers can be removed.

This change should only impact public builds, since the public additions
SDKs are what provide a SymlinkedHeaders.xcfilelist.

* Source/WTF/Scripts/symlink-public-sdk-headers-for-embedded-platforms:
(ln_sfh):
(framework_headers_cmp):
(framework_headers_cmp.diff):
(framework_copied_by_c_x_f_e_d): Deleted.
(framework_copied_by_c_x_f_e_d.diff): Deleted.

Canonical link: <a href="https://commits.webkit.org/282818@main">https://commits.webkit.org/282818@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ac7a4b6dc553aa7d4453c81f463e97c3cc171068

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/63800 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/43157 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/16397 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/67821 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/14408 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/51415 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/14688 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/51408 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/9959 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/66869 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/40414 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/55228 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/32025 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/37084 "Passed tests") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/12607 "Found 1 new test failure: imported/w3c/web-platform-tests/css/css-grid/subgrid/subgrid-baseline-006.html (failure)") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/13281 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/56912 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/59051 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/12934 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/69517 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/63045 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/7747 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/12487 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/58732 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/7780 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/55322 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/58879 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/14205 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/6440 "Passed tests") | [⏳ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/macOS-Ventura-Release-WK2-Intel-Tests-EWS "Waiting to run tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/84806 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/38977 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/14960 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/40056 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/41239 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/39799 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aproskuryakov reviewed (approved):
rs=me assuming that you did non-zero amount of testing.

We also need to do something about https://webkit.org/blog/3457/building-webkit-for-ios-simulator/

emw-apple committed:
[Build] Remove configure-xcode-for-embedded-development check in symlinking script
rdar://132780726
https://bugs.webkit.org/show_bug.cgi?id=277326

Reviewed by Alexey Proskuryakov.

Previously, this script checked for an Xcode installation that had been
modified by configure-xcode-for-embedded-development, and would re-use
those modified headers if needed. Since it has been a year since this
behavior change, and anyone buliding WebKit for embedded platforms needs
to be on a recent SDK version, this is no longer needed.

Instead, repurpose the check to detect when a header of framework we're
copying on top of the public SDK is no longer needed. Have WTF emit an
error indicating which headers can be removed.

This change should only impact public builds, since the public additions
SDKs are what provide a SymlinkedHeaders.xcfilelist.

* Source/WTF/Scripts/symlink-public-sdk-headers-for-embedded-platforms:
(ln_sfh):
(framework_headers_cmp):
(framework_headers_cmp.diff):
(framework_copied_by_c_x_f_e_d): Deleted.
(framework_copied_by_c_x_f_e_d.diff): Deleted.

Canonical link: https://commits.webkit.org/282818@main
